// ng-ai-chatbot/esm2015/ng-ai-chatbot.js
import { Injectable, Component, Input, Inject, NgModule } from '@angular/core';
import { HttpClient, HttpHeaders, HttpClientModule } from '@angular/common/http';
import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
import { MatDialogRef, MAT_DIALOG_DATA, MatDialog, MatToolbarModule, MatIconModule, MatCardModule, MatButtonModule, MatDialogModule, MatCheckboxModule, MatFormFieldModule, MatInputModule, MatListModule, MatGridListModule, MatSelectModule } from '@angular/material';
import { CommonModule } from '@angular/common';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class CoreService {
    /**
     * @param {?} http
     */
    constructor(http) {
        this.http = http;
    }
    /**
     * @param {?=} data
     * @return {?}
     */
    doHeaders(data = null) {
        const /** @type {?} */ headerVals = { 'Content-Type': 'application/json' };
        if (localStorage["botToken"]) {
            headerVals.Authorization = `Basic ${localStorage["botToken"]}`;
        }
        return new HttpHeaders(headerVals);
    }
    /**
     * @param {?} url
     * @param {?=} query
     * @param {?=} resJson
     * @return {?}
     */
    doGet(url, query = null, resJson = true) {
        return new Promise((resolve, reject) => {
            const /** @type {?} */ headers = this.doHeaders(null);
            const /** @type {?} */ finalUrl = this.doQueryString(CoreService.apiUrl + url, query);
            const /** @type {?} */ promise = this.http.get(finalUrl, { headers: headers });
            return promise.subscribe((data) => {
                if (data && data._body) {
                    resolve(JSON.parse(data._body));
                }
                else {
                    resolve(data);
                }
            }, error => {
                reject(error);
            });
        });
    }
    /**
     * @param {?} url
     * @param {?=} query
     * @param {?=} resJson
     * @return {?}
     */
    doDelete(url, query = null, resJson = true) {
        return new Promise((resolve, reject) => {
            const /** @type {?} */ headers = this.doHeaders(null);
            const /** @type {?} */ finalUrl = this.doQueryString(CoreService.apiUrl + url, query);
            const /** @type {?} */ promise = this.http.delete(finalUrl, { headers: headers });
            return promise.subscribe((data) => {
                if (data && data._body) {
                    resolve(JSON.parse(data._body));
                }
                else {
                    resolve(data);
                }
            }, error => {
                reject(error);
            });
        });
    }
    /**
     * @param {?} url
     * @param {?} data
     * @param {?=} files
     * @return {?}
     */
    doPost(url, data, files = null) {
        return this.doSubmit('post', url, data, files);
    }
    /**
     * @param {?} url
     * @param {?} data
     * @param {?=} files
     * @return {?}
     */
    doPut(url, data, files = null) {
        return this.doSubmit('put', url, data, files);
    }
    /**
     * @param {?} method
     * @param {?} url
     * @param {?} data
     * @param {?=} files
     * @return {?}
     */
    doSubmit(method, url, data, files = null) {
        return new Promise((resolve, reject) => {
            const /** @type {?} */ headers = this.doHeaders(data);
            if (!data) {
                data = {};
            }
            let /** @type {?} */ formData;
            if (files) {
                formData = new FormData();
                if (files.length === 1) {
                    formData.append('uploads', files[0], files[0].name);
                }
                else {
                    for (let /** @type {?} */ i = 0; i < files.length; i++) {
                        formData.append('uploads[]', files[i], files[i].name);
                    }
                }
            }
            else {
                formData = JSON.stringify(data);
            }
            console.log(`POST: ${CoreService.apiUrl + url}`);
            let /** @type {?} */ prom;
            if (method === 'post') {
                prom = this.http.post(CoreService.apiUrl + url, formData, { headers: headers });
            }
            else {
                prom = this.http.put(CoreService.apiUrl + url, formData, { headers: headers });
            }
            return prom
                .subscribe((dataAnswer) => {
                if (dataAnswer && dataAnswer._body) {
                    resolve(JSON.parse(dataAnswer._body));
                }
                else {
                    resolve(dataAnswer);
                }
            }, error => {
                reject(error);
            });
        });
    }
    /**
     * @param {?} url
     * @param {?} queries
     * @return {?}
     */
    doQueryString(url, queries) {
        if (queries) {
            url += '?';
            for (const /** @type {?} */ key in queries) {
                if (!key) {
                    continue;
                }
                url += `&${key}=${queries[key]}`;
            }
        }
        return url;
    }
    /**
     * @param {?} form
     * @param {?} keys
     * @param {?} data
     * @return {?}
     */
    setDataForm(form, keys, data) {
        if (!data) {
            return;
        }
        const /** @type {?} */ formData = {};
        for (const /** @type {?} */ key in keys) {
            if (key in data) {
                const /** @type {?} */ value = data[key];
                formData[key] = value;
            }
            else {
                formData[key] = '';
            }
        }
        // (<FormGroup>form).setValue(formData, { onlySelf: true });
        (/** @type {?} */ (form)).patchValue(formData, { onlySelf: true });
    }
}
CoreService.apiUrl = 'http://localhost:8001';
CoreService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
CoreService.ctorParameters = () => [
    { type: HttpClient, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class ChatService extends CoreService {
    /**
     * @param {?} http
     */
    constructor(http) {
        super(http);
        this.http = http;
    }
    /**
     * @param {?} intent
     * @param {?=} botId
     * @return {?}
     */
    send(intent, botId = 'default') {
        return this.doPost(`/api/v1?botId=${botId}`, intent);
    }
}
ChatService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
ChatService.ctorParameters = () => [
    { type: HttpClient, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class ChatComponent {
    /**
     * @param {?} fb
     * @param {?} chatService
     */
    constructor(fb, chatService) {
        this.fb = fb;
        this.chatService = chatService;
        this.chats = [];
        this.debug = false;
        this.chatFormFields = {
            input: [''],
        };
        this.chatForm = this.fb.group(this.chatFormFields);
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.chatInitial = {
            'currentNode': '',
            'complete': null, 'context': {},
            'parameters': [],
            'extractedParameters': {},
            'speechResponse': '',
            'intent': {},
            'input': 'init_conversation',
            'missingParameters': []
        };
        let /** @type {?} */ botId;
        if (this.bot && this.bot._id) {
            botId = this.bot._id.$oid;
        }
        this.chatService.send(this.chatInitial, botId)
            .then((c) => {
            c.owner = 'chat';
            this.changeCurrent(c);
        });
    }
    /**
     * @param {?} c
     * @return {?}
     */
    changeCurrent(c) {
        c.date = new Date();
        this.chats.push(c);
        this.chatCurrent = c;
        this.prettyChatCurrent = JSON ? JSON.stringify(c, null, '  ') : 'your browser doesnt support JSON so cant pretty print';
    }
    /**
     * @return {?}
     */
    send() {
        const /** @type {?} */ form = this.chatForm.value;
        const /** @type {?} */ sendMessage = Object.assign({}, this.chatCurrent, { input: form.input, owner: 'user' });
        this.changeCurrent(sendMessage);
        this.chatService.send(sendMessage)
            .then((c) => {
            c.owner = 'chat';
            this.changeCurrent(c);
            this.chatForm.reset();
        });
    }
}
ChatComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ng-ai-chatbot-chat',
                template: `<mat-card>
  {{title}}
  <div *ngFor="let chat of chats">
    <div class="row" *ngIf="chat.owner=='user'">
      <div class="col-md-10">
        <h4 class="textright fullwidth">{{chat.input}}</h4>
      </div>
      <div class="col-md-1" class="textcenter">
        <mat-icon> face </mat-icon>
      </div>
    </div>
    <div class="row" *ngIf="chat.owner=='chat'">
      <div class="col-md-1" class="textcenter">
        <mat-icon> android </mat-icon>
      </div>
      <div class="col-md-10">
        <h4 class="textleft fullwidth" [innerHTML]="chat.speechResponse"></h4>
      </div>
    </div>
    <!-- <p mat-line style="text-color: #ccc"> {{chat.date | date}} </p> -->
    <!-- <mat-divider></mat-divider> -->
  </div>
  <form [formGroup]="chatForm" *ngIf="chatForm" (submit)="send()">
    <mat-input-container>
      <input type="text" matInput formControlName="input" placeholder="Type your message here">
    </mat-input-container>
    <button mat-button>send</button>
  </form>
</mat-card>
<mat-card *ngIf="debug">
  POST /api/v1
  <pre>
    {{prettyChatCurrent}}
  </pre>
</mat-card>`,
                styles: [`.textright{
    text-align:right;
}
.textleft{
    text-align:left;
}
.textcenter{
    text-align:center;
}
.fullwidth{
    width:100%;
}`, `.container-fluid,
.container{
  margin-right:auto;
  margin-left:auto;
}
.container-fluid{
  padding-right:2rem;
  padding-left:2rem;
}
.row{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  display:-webkit-box;
  display:-ms-flexbox;
  display:flex;
  -webkit-box-flex:0;
  -ms-flex:0 1 auto;
  flex:0 1 auto;
  -webkit-box-orient:horizontal;
  -webkit-box-direction:normal;
  -ms-flex-direction:row;
  flex-direction:row;
  -ms-flex-wrap:wrap;
  flex-wrap:wrap;
  margin-right:-0.5rem;
  margin-left:-0.5rem;
}
.row.reverse{
  -webkit-box-orient:horizontal;
  -webkit-box-direction:reverse;
  -ms-flex-direction:row-reverse;
  flex-direction:row-reverse;
}
.col.reverse{
  -webkit-box-orient:vertical;
  -webkit-box-direction:reverse;
  -ms-flex-direction:column-reverse;
  flex-direction:column-reverse;
}
.col-xs,
.col-xs-1,
.col-xs-2,
.col-xs-3,
.col-xs-4,
.col-xs-5,
.col-xs-6,
.col-xs-7,
.col-xs-8,
.col-xs-9,
.col-xs-10,
.col-xs-11,
.col-xs-12,
.col-xs-offset-0,
.col-xs-offset-1,
.col-xs-offset-2,
.col-xs-offset-3,
.col-xs-offset-4,
.col-xs-offset-5,
.col-xs-offset-6,
.col-xs-offset-7,
.col-xs-offset-8,
.col-xs-offset-9,
.col-xs-offset-10,
.col-xs-offset-11,
.col-xs-offset-12{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  -webkit-box-flex:0;
  -ms-flex:0 0 auto;
  flex:0 0 auto;
  padding-right:0.5rem;
  padding-left:0.5rem;
}
.col-xs{
  -webkit-box-flex:1;
  -ms-flex-positive:1;
  flex-grow:1;
  -ms-flex-preferred-size:0;
  flex-basis:0;
  max-width:100%;
}
.col-xs-1{
  -ms-flex-preferred-size:8.33333333%;
  flex-basis:8.33333333%;
  max-width:8.33333333%;
}
.col-xs-2{
  -ms-flex-preferred-size:16.66666667%;
  flex-basis:16.66666667%;
  max-width:16.66666667%;
}
.col-xs-3{
  -ms-flex-preferred-size:25%;
  flex-basis:25%;
  max-width:25%;
}
.col-xs-4{
  -ms-flex-preferred-size:33.33333333%;
  flex-basis:33.33333333%;
  max-width:33.33333333%;
}
.col-xs-5{
  -ms-flex-preferred-size:41.66666667%;
  flex-basis:41.66666667%;
  max-width:41.66666667%;
}
.col-xs-6{
  -ms-flex-preferred-size:50%;
  flex-basis:50%;
  max-width:50%;
}
.col-xs-7{
  -ms-flex-preferred-size:58.33333333%;
  flex-basis:58.33333333%;
  max-width:58.33333333%;
}
.col-xs-8{
  -ms-flex-preferred-size:66.66666667%;
  flex-basis:66.66666667%;
  max-width:66.66666667%;
}
.col-xs-9{
  -ms-flex-preferred-size:75%;
  flex-basis:75%;
  max-width:75%;
}
.col-xs-10{
  -ms-flex-preferred-size:83.33333333%;
  flex-basis:83.33333333%;
  max-width:83.33333333%;
}
.col-xs-11{
  -ms-flex-preferred-size:91.66666667%;
  flex-basis:91.66666667%;
  max-width:91.66666667%;
}
.col-xs-12{
  -ms-flex-preferred-size:100%;
  flex-basis:100%;
  max-width:100%;
}
.col-xs-offset-0{
  margin-left:0;
}
.col-xs-offset-1{
  margin-left:8.33333333%;
}
.col-xs-offset-2{
  margin-left:16.66666667%;
}
.col-xs-offset-3{
  margin-left:25%;
}
.col-xs-offset-4{
  margin-left:33.33333333%;
}
.col-xs-offset-5{
  margin-left:41.66666667%;
}
.col-xs-offset-6{
  margin-left:50%;
}
.col-xs-offset-7{
  margin-left:58.33333333%;
}
.col-xs-offset-8{
  margin-left:66.66666667%;
}
.col-xs-offset-9{
  margin-left:75%;
}
.col-xs-offset-10{
  margin-left:83.33333333%;
}
.col-xs-offset-11{
  margin-left:91.66666667%;
}
.start-xs{
  -webkit-box-pack:start;
  -ms-flex-pack:start;
  justify-content:flex-start;
  text-align:start;
}
.center-xs{
  -webkit-box-pack:center;
  -ms-flex-pack:center;
  justify-content:center;
  text-align:center;
}
.end-xs{
  -webkit-box-pack:end;
  -ms-flex-pack:end;
  justify-content:flex-end;
  text-align:end;
}
.top-xs{
  -webkit-box-align:start;
  -ms-flex-align:start;
  align-items:flex-start;
}
.middle-xs{
  -webkit-box-align:center;
  -ms-flex-align:center;
  align-items:center;
}
.bottom-xs{
  -webkit-box-align:end;
  -ms-flex-align:end;
  align-items:flex-end;
}
.around-xs{
  -ms-flex-pack:distribute;
  justify-content:space-around;
}
.between-xs{
  -webkit-box-pack:justify;
  -ms-flex-pack:justify;
  justify-content:space-between;
}
.first-xs{
  -webkit-box-ordinal-group:0;
  -ms-flex-order:-1;
  order:-1;
}
.last-xs{
  -webkit-box-ordinal-group:2;
  -ms-flex-order:1;
  order:1;
}
@media only screen and (min-width: 48em){
  .container{
    width:49rem;
  }
  .col-sm,
  .col-sm-1,
  .col-sm-2,
  .col-sm-3,
  .col-sm-4,
  .col-sm-5,
  .col-sm-6,
  .col-sm-7,
  .col-sm-8,
  .col-sm-9,
  .col-sm-10,
  .col-sm-11,
  .col-sm-12,
  .col-sm-offset-0,
  .col-sm-offset-1,
  .col-sm-offset-2,
  .col-sm-offset-3,
  .col-sm-offset-4,
  .col-sm-offset-5,
  .col-sm-offset-6,
  .col-sm-offset-7,
  .col-sm-offset-8,
  .col-sm-offset-9,
  .col-sm-offset-10,
  .col-sm-offset-11,
  .col-sm-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-sm{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-sm-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-sm-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-sm-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-sm-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-sm-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-sm-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-sm-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-sm-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-sm-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-sm-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-sm-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-sm-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-sm-offset-0{
    margin-left:0;
  }
  .col-sm-offset-1{
    margin-left:8.33333333%;
  }
  .col-sm-offset-2{
    margin-left:16.66666667%;
  }
  .col-sm-offset-3{
    margin-left:25%;
  }
  .col-sm-offset-4{
    margin-left:33.33333333%;
  }
  .col-sm-offset-5{
    margin-left:41.66666667%;
  }
  .col-sm-offset-6{
    margin-left:50%;
  }
  .col-sm-offset-7{
    margin-left:58.33333333%;
  }
  .col-sm-offset-8{
    margin-left:66.66666667%;
  }
  .col-sm-offset-9{
    margin-left:75%;
  }
  .col-sm-offset-10{
    margin-left:83.33333333%;
  }
  .col-sm-offset-11{
    margin-left:91.66666667%;
  }
  .start-sm{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-sm{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-sm{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-sm{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-sm{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-sm{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-sm{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-sm{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-sm{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-sm{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 64em){
  .container{
    width:65rem;
  }
  .col-md,
  .col-md-1,
  .col-md-2,
  .col-md-3,
  .col-md-4,
  .col-md-5,
  .col-md-6,
  .col-md-7,
  .col-md-8,
  .col-md-9,
  .col-md-10,
  .col-md-11,
  .col-md-12,
  .col-md-offset-0,
  .col-md-offset-1,
  .col-md-offset-2,
  .col-md-offset-3,
  .col-md-offset-4,
  .col-md-offset-5,
  .col-md-offset-6,
  .col-md-offset-7,
  .col-md-offset-8,
  .col-md-offset-9,
  .col-md-offset-10,
  .col-md-offset-11,
  .col-md-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-md{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-md-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-md-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-md-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-md-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-md-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-md-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-md-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-md-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-md-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-md-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-md-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-md-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-md-offset-0{
    margin-left:0;
  }
  .col-md-offset-1{
    margin-left:8.33333333%;
  }
  .col-md-offset-2{
    margin-left:16.66666667%;
  }
  .col-md-offset-3{
    margin-left:25%;
  }
  .col-md-offset-4{
    margin-left:33.33333333%;
  }
  .col-md-offset-5{
    margin-left:41.66666667%;
  }
  .col-md-offset-6{
    margin-left:50%;
  }
  .col-md-offset-7{
    margin-left:58.33333333%;
  }
  .col-md-offset-8{
    margin-left:66.66666667%;
  }
  .col-md-offset-9{
    margin-left:75%;
  }
  .col-md-offset-10{
    margin-left:83.33333333%;
  }
  .col-md-offset-11{
    margin-left:91.66666667%;
  }
  .start-md{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-md{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-md{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-md{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-md{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-md{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-md{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-md{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-md{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-md{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 75em){
  .container{
    width:76rem;
  }
  .col-lg,
  .col-lg-1,
  .col-lg-2,
  .col-lg-3,
  .col-lg-4,
  .col-lg-5,
  .col-lg-6,
  .col-lg-7,
  .col-lg-8,
  .col-lg-9,
  .col-lg-10,
  .col-lg-11,
  .col-lg-12,
  .col-lg-offset-0,
  .col-lg-offset-1,
  .col-lg-offset-2,
  .col-lg-offset-3,
  .col-lg-offset-4,
  .col-lg-offset-5,
  .col-lg-offset-6,
  .col-lg-offset-7,
  .col-lg-offset-8,
  .col-lg-offset-9,
  .col-lg-offset-10,
  .col-lg-offset-11,
  .col-lg-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-lg{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-lg-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-lg-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-lg-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-lg-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-lg-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-lg-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-lg-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-lg-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-lg-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-lg-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-lg-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-lg-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-lg-offset-0{
    margin-left:0;
  }
  .col-lg-offset-1{
    margin-left:8.33333333%;
  }
  .col-lg-offset-2{
    margin-left:16.66666667%;
  }
  .col-lg-offset-3{
    margin-left:25%;
  }
  .col-lg-offset-4{
    margin-left:33.33333333%;
  }
  .col-lg-offset-5{
    margin-left:41.66666667%;
  }
  .col-lg-offset-6{
    margin-left:50%;
  }
  .col-lg-offset-7{
    margin-left:58.33333333%;
  }
  .col-lg-offset-8{
    margin-left:66.66666667%;
  }
  .col-lg-offset-9{
    margin-left:75%;
  }
  .col-lg-offset-10{
    margin-left:83.33333333%;
  }
  .col-lg-offset-11{
    margin-left:91.66666667%;
  }
  .start-lg{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-lg{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-lg{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-lg{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-lg{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-lg{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-lg{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-lg{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-lg{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-lg{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}`]
            },] },
];
/** @nocollapse */
ChatComponent.ctorParameters = () => [
    { type: FormBuilder, },
    { type: ChatService, },
];
ChatComponent.propDecorators = {
    "bot": [{ type: Input },],
    "debug": [{ type: Input },],
    "title": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class StoryService extends CoreService {
    /**
     * @param {?} http
     */
    constructor(http) {
        super(http);
        this.http = http;
    }
    /**
     * @param {?} botId
     * @return {?}
     */
    getStories(botId) {
        let /** @type {?} */ url = '/stories/';
        if (botId) {
            url = `${url}?botId=${botId}`;
        }
        return this.doGet(url);
    }
    /**
     * @param {?} id
     * @return {?}
     */
    getStory(id) {
        return this.doGet(`/stories/edit/${id}`);
    }
    /**
     * @param {?} story
     * @return {?}
     */
    saveStory(story) {
        if (story._id) {
            return this.updateStory(story);
        }
        else {
            delete story._id;
            return this.createStory(story);
        }
    }
    /**
     * @param {?} story
     * @return {?}
     */
    createStory(story) {
        return this.doPost(`/stories/`, story);
    }
    /**
     * @param {?} story
     * @return {?}
     */
    updateStory(story) {
        return this.doPut(`/stories/${story._id}`, story);
    }
    /**
     * @param {?} id
     * @return {?}
     */
    buildStory(id) {
        return this.doPost(`/core/buildModel/${id}`, {});
    }
    /**
     * @param {?} id
     * @return {?}
     */
    deleteStory(id) {
        return this.doDelete(`/stories/${id}`, {});
    }
    /**
     *
     * @param {?} sentences ex.: i'm searching product
     * @return {?}
     */
    startLabeling(sentences) {
        return this.doPost(`/core/posTagAndLabel?json=true`, { sentences });
    }
    /**
     *
     * @param {?} storyId  ex.:59fddff51ec5e81bf9d6e021
     * @param {?} labeledSentence   ex.: [["i","NN","O"],["'m","VBP","O"],["searching","VBG","O"],["product","NN","O"]]
     * @param {?=} botId
     * @return {?}
     */
    addToTestSet(storyId, labeledSentence, botId = 'default') {
        return this.doPost(`/train/insertLabeledSentence`, { storyId, botId, labeledSentence });
    }
    /**
     * @return {?}
     */
    getBots() {
        return this.doGet(`/stories/bot`);
    }
    /**
     * @param {?} bot
     * @return {?}
     */
    saveBot(bot) {
        if (bot._id) {
            return this.updateBot(bot);
        }
        else {
            delete bot._id;
            return this.createBot(bot);
        }
    }
    /**
     * @param {?} story
     * @return {?}
     */
    createBot(story) {
        return this.doPost(`/stories/bot`, story);
    }
    /**
     * @param {?} story
     * @return {?}
     */
    updateBot(story) {
        return this.doPut(`/stories/bot/${story._id}`, story);
    }
}
StoryService.storyTypes = {
    'mobile': 'Mobile number',
    'email': 'Email',
    'free_text': 'Free Text',
    'number': 'Number',
    'list': 'List',
};
StoryService.decorators = [
    { type: Injectable },
];
/** @nocollapse */
StoryService.ctorParameters = () => [
    { type: HttpClient, },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class StoryDialogComponent {
    /**
     * @param {?} dialogRef
     * @param {?} data
     */
    constructor(dialogRef, data) {
        this.dialogRef = dialogRef;
        this.data = data;
        if (data.story) {
            this.story = data.story;
        }
    }
    /**
     * @param {?} response
     * @return {?}
     */
    close(response) {
        this.dialogRef.close();
    }
}
StoryDialogComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ng-ai-chatbot-storydialog',
                template: '<ng-ai-chatbot-story [story]="story" [afterSave]="close.bind(this)"></ng-ai-chatbot-story>',
            },] },
];
/** @nocollapse */
StoryDialogComponent.ctorParameters = () => [
    { type: MatDialogRef, },
    { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class TrainDialogComponent {
    /**
     * @param {?} dialogRef
     * @param {?} data
     */
    constructor(dialogRef, data) {
        this.dialogRef = dialogRef;
        this.data = data;
        if (data && data.story) {
            this.story = data.story;
        }
    }
    /**
     * @return {?}
     */
    close() {
        this.dialogRef.close();
    }
}
TrainDialogComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ng-ai-chatbot-traindialog',
                template: '<ng-ai-chatbot-train [story]="story"></ng-ai-chatbot-train>',
            },] },
];
/** @nocollapse */
TrainDialogComponent.ctorParameters = () => [
    { type: MatDialogRef, },
    { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] },] },
];
TrainDialogComponent.propDecorators = {
    "story": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class StoriesComponent {
    /**
     * @param {?} dialog
     * @param {?} storyService
     */
    constructor(dialog, storyService) {
        this.dialog = dialog;
        this.storyService = storyService;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        let /** @type {?} */ botId;
        if (this.bot && this.bot._id) {
            botId = this.bot._id.$oid;
        }
        this.storyService.getStories(botId).then((s) => {
            this.stories = s;
        });
    }
    /**
     * @param {?=} story
     * @return {?}
     */
    openDialog(story = null) {
        const /** @type {?} */ dialogRef = this.dialog.open(StoryDialogComponent, {
            width: '90%',
            data: { story: story }
        });
        dialogRef.afterClosed().subscribe(result => {
            console.log('The dialog was closed');
            console.log(result);
            this.ngOnInit();
        });
    }
    /**
     * @return {?}
     */
    add() {
        this.openDialog();
    }
    /**
     * @param {?} story
     * @return {?}
     */
    edit(story) {
        this.openDialog(story);
    }
    /**
     * @param {?} story
     * @return {?}
     */
    train(story) {
        const /** @type {?} */ dialogRef = this.dialog.open(TrainDialogComponent, {
            width: '99%',
            data: { story: story }
        });
        dialogRef.afterClosed().subscribe(result => {
            console.log('The dialog was closed');
            console.log(result);
        });
    }
    /**
     * @param {?} story
     * @return {?}
     */
    delete(story) {
        if (confirm('Are u sure want to delete this story?')) {
            this.storyService.deleteStory(story._id.$oid).then((s) => {
                this.ngOnInit();
            });
        }
    }
    /**
     * @param {?} story
     * @return {?}
     */
    build(story) {
        this.storyService.buildStory(story._id.$oid).then((s) => {
            this.ngOnInit();
        });
    }
}
StoriesComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ng-ai-chatbot-stories',
                template: `
<h1>
  Stories
  <span *ngIf="bot"> for {{bot.botName}}</span>
  <button mat-mini-fab (click)="add()">
    <mat-icon aria-label="Example icon-button with a heart icon">add</mat-icon>
  </button>
</h1>
<div class="row">
  <mat-card class="col-md-6" *ngFor="let story of stories">
    <mat-card-header>
      <div mat-card-avatar class="example-header-image"></div>
      <mat-card-title>{{story.storyName}}</mat-card-title>
    </mat-card-header>
    <mat-card-actions>
      <button mat-button (click)="edit(story)" >Edit</button>
      <button mat-button (click)="train(story)" >Train</button>
      <button mat-button (click)="delete(story)" >Delete</button>
      <button mat-button (click)="build(story)" >Build</button>
    </mat-card-actions>
  </mat-card>
</div>
`,
                styles: [``, `.container-fluid,
.container{
  margin-right:auto;
  margin-left:auto;
}
.container-fluid{
  padding-right:2rem;
  padding-left:2rem;
}
.row{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  display:-webkit-box;
  display:-ms-flexbox;
  display:flex;
  -webkit-box-flex:0;
  -ms-flex:0 1 auto;
  flex:0 1 auto;
  -webkit-box-orient:horizontal;
  -webkit-box-direction:normal;
  -ms-flex-direction:row;
  flex-direction:row;
  -ms-flex-wrap:wrap;
  flex-wrap:wrap;
  margin-right:-0.5rem;
  margin-left:-0.5rem;
}
.row.reverse{
  -webkit-box-orient:horizontal;
  -webkit-box-direction:reverse;
  -ms-flex-direction:row-reverse;
  flex-direction:row-reverse;
}
.col.reverse{
  -webkit-box-orient:vertical;
  -webkit-box-direction:reverse;
  -ms-flex-direction:column-reverse;
  flex-direction:column-reverse;
}
.col-xs,
.col-xs-1,
.col-xs-2,
.col-xs-3,
.col-xs-4,
.col-xs-5,
.col-xs-6,
.col-xs-7,
.col-xs-8,
.col-xs-9,
.col-xs-10,
.col-xs-11,
.col-xs-12,
.col-xs-offset-0,
.col-xs-offset-1,
.col-xs-offset-2,
.col-xs-offset-3,
.col-xs-offset-4,
.col-xs-offset-5,
.col-xs-offset-6,
.col-xs-offset-7,
.col-xs-offset-8,
.col-xs-offset-9,
.col-xs-offset-10,
.col-xs-offset-11,
.col-xs-offset-12{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  -webkit-box-flex:0;
  -ms-flex:0 0 auto;
  flex:0 0 auto;
  padding-right:0.5rem;
  padding-left:0.5rem;
}
.col-xs{
  -webkit-box-flex:1;
  -ms-flex-positive:1;
  flex-grow:1;
  -ms-flex-preferred-size:0;
  flex-basis:0;
  max-width:100%;
}
.col-xs-1{
  -ms-flex-preferred-size:8.33333333%;
  flex-basis:8.33333333%;
  max-width:8.33333333%;
}
.col-xs-2{
  -ms-flex-preferred-size:16.66666667%;
  flex-basis:16.66666667%;
  max-width:16.66666667%;
}
.col-xs-3{
  -ms-flex-preferred-size:25%;
  flex-basis:25%;
  max-width:25%;
}
.col-xs-4{
  -ms-flex-preferred-size:33.33333333%;
  flex-basis:33.33333333%;
  max-width:33.33333333%;
}
.col-xs-5{
  -ms-flex-preferred-size:41.66666667%;
  flex-basis:41.66666667%;
  max-width:41.66666667%;
}
.col-xs-6{
  -ms-flex-preferred-size:50%;
  flex-basis:50%;
  max-width:50%;
}
.col-xs-7{
  -ms-flex-preferred-size:58.33333333%;
  flex-basis:58.33333333%;
  max-width:58.33333333%;
}
.col-xs-8{
  -ms-flex-preferred-size:66.66666667%;
  flex-basis:66.66666667%;
  max-width:66.66666667%;
}
.col-xs-9{
  -ms-flex-preferred-size:75%;
  flex-basis:75%;
  max-width:75%;
}
.col-xs-10{
  -ms-flex-preferred-size:83.33333333%;
  flex-basis:83.33333333%;
  max-width:83.33333333%;
}
.col-xs-11{
  -ms-flex-preferred-size:91.66666667%;
  flex-basis:91.66666667%;
  max-width:91.66666667%;
}
.col-xs-12{
  -ms-flex-preferred-size:100%;
  flex-basis:100%;
  max-width:100%;
}
.col-xs-offset-0{
  margin-left:0;
}
.col-xs-offset-1{
  margin-left:8.33333333%;
}
.col-xs-offset-2{
  margin-left:16.66666667%;
}
.col-xs-offset-3{
  margin-left:25%;
}
.col-xs-offset-4{
  margin-left:33.33333333%;
}
.col-xs-offset-5{
  margin-left:41.66666667%;
}
.col-xs-offset-6{
  margin-left:50%;
}
.col-xs-offset-7{
  margin-left:58.33333333%;
}
.col-xs-offset-8{
  margin-left:66.66666667%;
}
.col-xs-offset-9{
  margin-left:75%;
}
.col-xs-offset-10{
  margin-left:83.33333333%;
}
.col-xs-offset-11{
  margin-left:91.66666667%;
}
.start-xs{
  -webkit-box-pack:start;
  -ms-flex-pack:start;
  justify-content:flex-start;
  text-align:start;
}
.center-xs{
  -webkit-box-pack:center;
  -ms-flex-pack:center;
  justify-content:center;
  text-align:center;
}
.end-xs{
  -webkit-box-pack:end;
  -ms-flex-pack:end;
  justify-content:flex-end;
  text-align:end;
}
.top-xs{
  -webkit-box-align:start;
  -ms-flex-align:start;
  align-items:flex-start;
}
.middle-xs{
  -webkit-box-align:center;
  -ms-flex-align:center;
  align-items:center;
}
.bottom-xs{
  -webkit-box-align:end;
  -ms-flex-align:end;
  align-items:flex-end;
}
.around-xs{
  -ms-flex-pack:distribute;
  justify-content:space-around;
}
.between-xs{
  -webkit-box-pack:justify;
  -ms-flex-pack:justify;
  justify-content:space-between;
}
.first-xs{
  -webkit-box-ordinal-group:0;
  -ms-flex-order:-1;
  order:-1;
}
.last-xs{
  -webkit-box-ordinal-group:2;
  -ms-flex-order:1;
  order:1;
}
@media only screen and (min-width: 48em){
  .container{
    width:49rem;
  }
  .col-sm,
  .col-sm-1,
  .col-sm-2,
  .col-sm-3,
  .col-sm-4,
  .col-sm-5,
  .col-sm-6,
  .col-sm-7,
  .col-sm-8,
  .col-sm-9,
  .col-sm-10,
  .col-sm-11,
  .col-sm-12,
  .col-sm-offset-0,
  .col-sm-offset-1,
  .col-sm-offset-2,
  .col-sm-offset-3,
  .col-sm-offset-4,
  .col-sm-offset-5,
  .col-sm-offset-6,
  .col-sm-offset-7,
  .col-sm-offset-8,
  .col-sm-offset-9,
  .col-sm-offset-10,
  .col-sm-offset-11,
  .col-sm-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-sm{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-sm-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-sm-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-sm-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-sm-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-sm-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-sm-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-sm-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-sm-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-sm-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-sm-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-sm-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-sm-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-sm-offset-0{
    margin-left:0;
  }
  .col-sm-offset-1{
    margin-left:8.33333333%;
  }
  .col-sm-offset-2{
    margin-left:16.66666667%;
  }
  .col-sm-offset-3{
    margin-left:25%;
  }
  .col-sm-offset-4{
    margin-left:33.33333333%;
  }
  .col-sm-offset-5{
    margin-left:41.66666667%;
  }
  .col-sm-offset-6{
    margin-left:50%;
  }
  .col-sm-offset-7{
    margin-left:58.33333333%;
  }
  .col-sm-offset-8{
    margin-left:66.66666667%;
  }
  .col-sm-offset-9{
    margin-left:75%;
  }
  .col-sm-offset-10{
    margin-left:83.33333333%;
  }
  .col-sm-offset-11{
    margin-left:91.66666667%;
  }
  .start-sm{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-sm{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-sm{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-sm{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-sm{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-sm{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-sm{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-sm{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-sm{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-sm{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 64em){
  .container{
    width:65rem;
  }
  .col-md,
  .col-md-1,
  .col-md-2,
  .col-md-3,
  .col-md-4,
  .col-md-5,
  .col-md-6,
  .col-md-7,
  .col-md-8,
  .col-md-9,
  .col-md-10,
  .col-md-11,
  .col-md-12,
  .col-md-offset-0,
  .col-md-offset-1,
  .col-md-offset-2,
  .col-md-offset-3,
  .col-md-offset-4,
  .col-md-offset-5,
  .col-md-offset-6,
  .col-md-offset-7,
  .col-md-offset-8,
  .col-md-offset-9,
  .col-md-offset-10,
  .col-md-offset-11,
  .col-md-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-md{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-md-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-md-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-md-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-md-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-md-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-md-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-md-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-md-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-md-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-md-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-md-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-md-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-md-offset-0{
    margin-left:0;
  }
  .col-md-offset-1{
    margin-left:8.33333333%;
  }
  .col-md-offset-2{
    margin-left:16.66666667%;
  }
  .col-md-offset-3{
    margin-left:25%;
  }
  .col-md-offset-4{
    margin-left:33.33333333%;
  }
  .col-md-offset-5{
    margin-left:41.66666667%;
  }
  .col-md-offset-6{
    margin-left:50%;
  }
  .col-md-offset-7{
    margin-left:58.33333333%;
  }
  .col-md-offset-8{
    margin-left:66.66666667%;
  }
  .col-md-offset-9{
    margin-left:75%;
  }
  .col-md-offset-10{
    margin-left:83.33333333%;
  }
  .col-md-offset-11{
    margin-left:91.66666667%;
  }
  .start-md{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-md{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-md{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-md{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-md{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-md{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-md{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-md{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-md{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-md{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 75em){
  .container{
    width:76rem;
  }
  .col-lg,
  .col-lg-1,
  .col-lg-2,
  .col-lg-3,
  .col-lg-4,
  .col-lg-5,
  .col-lg-6,
  .col-lg-7,
  .col-lg-8,
  .col-lg-9,
  .col-lg-10,
  .col-lg-11,
  .col-lg-12,
  .col-lg-offset-0,
  .col-lg-offset-1,
  .col-lg-offset-2,
  .col-lg-offset-3,
  .col-lg-offset-4,
  .col-lg-offset-5,
  .col-lg-offset-6,
  .col-lg-offset-7,
  .col-lg-offset-8,
  .col-lg-offset-9,
  .col-lg-offset-10,
  .col-lg-offset-11,
  .col-lg-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-lg{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-lg-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-lg-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-lg-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-lg-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-lg-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-lg-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-lg-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-lg-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-lg-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-lg-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-lg-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-lg-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-lg-offset-0{
    margin-left:0;
  }
  .col-lg-offset-1{
    margin-left:8.33333333%;
  }
  .col-lg-offset-2{
    margin-left:16.66666667%;
  }
  .col-lg-offset-3{
    margin-left:25%;
  }
  .col-lg-offset-4{
    margin-left:33.33333333%;
  }
  .col-lg-offset-5{
    margin-left:41.66666667%;
  }
  .col-lg-offset-6{
    margin-left:50%;
  }
  .col-lg-offset-7{
    margin-left:58.33333333%;
  }
  .col-lg-offset-8{
    margin-left:66.66666667%;
  }
  .col-lg-offset-9{
    margin-left:75%;
  }
  .col-lg-offset-10{
    margin-left:83.33333333%;
  }
  .col-lg-offset-11{
    margin-left:91.66666667%;
  }
  .start-lg{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-lg{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-lg{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-lg{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-lg{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-lg{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-lg{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-lg{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-lg{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-lg{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}`]
            },] },
];
/** @nocollapse */
StoriesComponent.ctorParameters = () => [
    { type: MatDialog, },
    { type: StoryService, },
];
StoriesComponent.propDecorators = {
    "bot": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class TrainComponent {
    /**
     * @param {?} fb
     * @param {?} storyService
     */
    constructor(fb, storyService) {
        this.fb = fb;
        this.storyService = storyService;
        this.trainFormFields = {
            input: [''],
        };
        this.trainForm = this.fb.group(this.trainFormFields);
        this.testFormFields = {
            input: [''],
        };
        this.testForm = this.fb.group(this.testFormFields);
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        console.log(this.story);
        if (this.story) {
            if (this.story._id && this.story._id.$oid) {
                this.story._id = this.story._id.$oid;
            }
        }
    }
    /**
     * @return {?}
     */
    addTest() {
    }
    /**
     * @return {?}
     */
    buildModel() {
        this.storyService.buildStory(this.story._id)
            .then(c => {
            this.message = 'Build sucessfull';
        })
            .catch(c => {
            this.message = 'Error on Building';
        });
    }
    /**
     * @return {?}
     */
    clear() {
        this.trainForm.reset();
    }
    /**
     * @return {?}
     */
    startLabeling() {
        const /** @type {?} */ form = this.trainForm.value;
        this.storyService.startLabeling(form.input)
            .then(c => {
            this.labelled = c;
            // this.message = 'Build sucessfull';
        })
            .catch(c => {
            this.message = 'Error on labelling';
        });
    }
    /**
     * @param {?} i
     * @return {?}
     */
    deleteSetence(i) {
        const /** @type {?} */ array = this.story.labeledSentences;
        array.splice(i, 1);
    }
}
TrainComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ng-ai-chatbot-train',
                template: `<p>
  Train your story for possible user inputs. Tag required parameters using mouse and give em labels
</p>
<form [formGroup]="trainForm" *ngIf="trainForm" (submit)="startLabeling()" class="row">
  <mat-input-container class="col-md-10">
    <textarea matInput rows="3" formControlName="input" placeholder="Enter a sample input"></textarea>
  </mat-input-container>
  <button class="col-md-1" type="button" (click)="clear()" mat-button>Clear</button>
  <button class="col-md-1" mat-button>Start Labeling</button>
</form>
<form [formGroup]="testForm" *ngIf="testForm" (submit)="addTest()" class="row">
  <mat-input-container class="col-md-8">
    <input type="text" matInput formControlName="input" placeholder="Mark as label">
  </mat-input-container>
  <button type="button" class="col-md-2" (click)="buildModel()" mat-button>Build Model</button>
  <button mat-button class="col-md-2">Add To Test Set</button>
</form>
<p *ngIf="labelled">
  {{labelled}}
</p>
<h2>View labeled sentences</h2>
<section class="labeledSentences" *ngIf="story">
  <div *ngFor="let trainS of story.labeledSentences; let i=index;" class="row">
    <p clas="col-md-10">{{trainS.data}}
    </p>
    <button type="button" class="col-md-2" (click)="deleteSetence(i)" mat-button>delete</button>
  </div>
</section>`,
                styles: [`
section .labeledSentences{
    overflow-y:auto;
    height:200px;
}`, `.container-fluid,
.container{
  margin-right:auto;
  margin-left:auto;
}
.container-fluid{
  padding-right:2rem;
  padding-left:2rem;
}
.row{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  display:-webkit-box;
  display:-ms-flexbox;
  display:flex;
  -webkit-box-flex:0;
  -ms-flex:0 1 auto;
  flex:0 1 auto;
  -webkit-box-orient:horizontal;
  -webkit-box-direction:normal;
  -ms-flex-direction:row;
  flex-direction:row;
  -ms-flex-wrap:wrap;
  flex-wrap:wrap;
  margin-right:-0.5rem;
  margin-left:-0.5rem;
}
.row.reverse{
  -webkit-box-orient:horizontal;
  -webkit-box-direction:reverse;
  -ms-flex-direction:row-reverse;
  flex-direction:row-reverse;
}
.col.reverse{
  -webkit-box-orient:vertical;
  -webkit-box-direction:reverse;
  -ms-flex-direction:column-reverse;
  flex-direction:column-reverse;
}
.col-xs,
.col-xs-1,
.col-xs-2,
.col-xs-3,
.col-xs-4,
.col-xs-5,
.col-xs-6,
.col-xs-7,
.col-xs-8,
.col-xs-9,
.col-xs-10,
.col-xs-11,
.col-xs-12,
.col-xs-offset-0,
.col-xs-offset-1,
.col-xs-offset-2,
.col-xs-offset-3,
.col-xs-offset-4,
.col-xs-offset-5,
.col-xs-offset-6,
.col-xs-offset-7,
.col-xs-offset-8,
.col-xs-offset-9,
.col-xs-offset-10,
.col-xs-offset-11,
.col-xs-offset-12{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  -webkit-box-flex:0;
  -ms-flex:0 0 auto;
  flex:0 0 auto;
  padding-right:0.5rem;
  padding-left:0.5rem;
}
.col-xs{
  -webkit-box-flex:1;
  -ms-flex-positive:1;
  flex-grow:1;
  -ms-flex-preferred-size:0;
  flex-basis:0;
  max-width:100%;
}
.col-xs-1{
  -ms-flex-preferred-size:8.33333333%;
  flex-basis:8.33333333%;
  max-width:8.33333333%;
}
.col-xs-2{
  -ms-flex-preferred-size:16.66666667%;
  flex-basis:16.66666667%;
  max-width:16.66666667%;
}
.col-xs-3{
  -ms-flex-preferred-size:25%;
  flex-basis:25%;
  max-width:25%;
}
.col-xs-4{
  -ms-flex-preferred-size:33.33333333%;
  flex-basis:33.33333333%;
  max-width:33.33333333%;
}
.col-xs-5{
  -ms-flex-preferred-size:41.66666667%;
  flex-basis:41.66666667%;
  max-width:41.66666667%;
}
.col-xs-6{
  -ms-flex-preferred-size:50%;
  flex-basis:50%;
  max-width:50%;
}
.col-xs-7{
  -ms-flex-preferred-size:58.33333333%;
  flex-basis:58.33333333%;
  max-width:58.33333333%;
}
.col-xs-8{
  -ms-flex-preferred-size:66.66666667%;
  flex-basis:66.66666667%;
  max-width:66.66666667%;
}
.col-xs-9{
  -ms-flex-preferred-size:75%;
  flex-basis:75%;
  max-width:75%;
}
.col-xs-10{
  -ms-flex-preferred-size:83.33333333%;
  flex-basis:83.33333333%;
  max-width:83.33333333%;
}
.col-xs-11{
  -ms-flex-preferred-size:91.66666667%;
  flex-basis:91.66666667%;
  max-width:91.66666667%;
}
.col-xs-12{
  -ms-flex-preferred-size:100%;
  flex-basis:100%;
  max-width:100%;
}
.col-xs-offset-0{
  margin-left:0;
}
.col-xs-offset-1{
  margin-left:8.33333333%;
}
.col-xs-offset-2{
  margin-left:16.66666667%;
}
.col-xs-offset-3{
  margin-left:25%;
}
.col-xs-offset-4{
  margin-left:33.33333333%;
}
.col-xs-offset-5{
  margin-left:41.66666667%;
}
.col-xs-offset-6{
  margin-left:50%;
}
.col-xs-offset-7{
  margin-left:58.33333333%;
}
.col-xs-offset-8{
  margin-left:66.66666667%;
}
.col-xs-offset-9{
  margin-left:75%;
}
.col-xs-offset-10{
  margin-left:83.33333333%;
}
.col-xs-offset-11{
  margin-left:91.66666667%;
}
.start-xs{
  -webkit-box-pack:start;
  -ms-flex-pack:start;
  justify-content:flex-start;
  text-align:start;
}
.center-xs{
  -webkit-box-pack:center;
  -ms-flex-pack:center;
  justify-content:center;
  text-align:center;
}
.end-xs{
  -webkit-box-pack:end;
  -ms-flex-pack:end;
  justify-content:flex-end;
  text-align:end;
}
.top-xs{
  -webkit-box-align:start;
  -ms-flex-align:start;
  align-items:flex-start;
}
.middle-xs{
  -webkit-box-align:center;
  -ms-flex-align:center;
  align-items:center;
}
.bottom-xs{
  -webkit-box-align:end;
  -ms-flex-align:end;
  align-items:flex-end;
}
.around-xs{
  -ms-flex-pack:distribute;
  justify-content:space-around;
}
.between-xs{
  -webkit-box-pack:justify;
  -ms-flex-pack:justify;
  justify-content:space-between;
}
.first-xs{
  -webkit-box-ordinal-group:0;
  -ms-flex-order:-1;
  order:-1;
}
.last-xs{
  -webkit-box-ordinal-group:2;
  -ms-flex-order:1;
  order:1;
}
@media only screen and (min-width: 48em){
  .container{
    width:49rem;
  }
  .col-sm,
  .col-sm-1,
  .col-sm-2,
  .col-sm-3,
  .col-sm-4,
  .col-sm-5,
  .col-sm-6,
  .col-sm-7,
  .col-sm-8,
  .col-sm-9,
  .col-sm-10,
  .col-sm-11,
  .col-sm-12,
  .col-sm-offset-0,
  .col-sm-offset-1,
  .col-sm-offset-2,
  .col-sm-offset-3,
  .col-sm-offset-4,
  .col-sm-offset-5,
  .col-sm-offset-6,
  .col-sm-offset-7,
  .col-sm-offset-8,
  .col-sm-offset-9,
  .col-sm-offset-10,
  .col-sm-offset-11,
  .col-sm-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-sm{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-sm-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-sm-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-sm-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-sm-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-sm-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-sm-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-sm-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-sm-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-sm-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-sm-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-sm-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-sm-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-sm-offset-0{
    margin-left:0;
  }
  .col-sm-offset-1{
    margin-left:8.33333333%;
  }
  .col-sm-offset-2{
    margin-left:16.66666667%;
  }
  .col-sm-offset-3{
    margin-left:25%;
  }
  .col-sm-offset-4{
    margin-left:33.33333333%;
  }
  .col-sm-offset-5{
    margin-left:41.66666667%;
  }
  .col-sm-offset-6{
    margin-left:50%;
  }
  .col-sm-offset-7{
    margin-left:58.33333333%;
  }
  .col-sm-offset-8{
    margin-left:66.66666667%;
  }
  .col-sm-offset-9{
    margin-left:75%;
  }
  .col-sm-offset-10{
    margin-left:83.33333333%;
  }
  .col-sm-offset-11{
    margin-left:91.66666667%;
  }
  .start-sm{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-sm{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-sm{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-sm{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-sm{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-sm{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-sm{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-sm{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-sm{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-sm{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 64em){
  .container{
    width:65rem;
  }
  .col-md,
  .col-md-1,
  .col-md-2,
  .col-md-3,
  .col-md-4,
  .col-md-5,
  .col-md-6,
  .col-md-7,
  .col-md-8,
  .col-md-9,
  .col-md-10,
  .col-md-11,
  .col-md-12,
  .col-md-offset-0,
  .col-md-offset-1,
  .col-md-offset-2,
  .col-md-offset-3,
  .col-md-offset-4,
  .col-md-offset-5,
  .col-md-offset-6,
  .col-md-offset-7,
  .col-md-offset-8,
  .col-md-offset-9,
  .col-md-offset-10,
  .col-md-offset-11,
  .col-md-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-md{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-md-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-md-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-md-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-md-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-md-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-md-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-md-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-md-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-md-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-md-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-md-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-md-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-md-offset-0{
    margin-left:0;
  }
  .col-md-offset-1{
    margin-left:8.33333333%;
  }
  .col-md-offset-2{
    margin-left:16.66666667%;
  }
  .col-md-offset-3{
    margin-left:25%;
  }
  .col-md-offset-4{
    margin-left:33.33333333%;
  }
  .col-md-offset-5{
    margin-left:41.66666667%;
  }
  .col-md-offset-6{
    margin-left:50%;
  }
  .col-md-offset-7{
    margin-left:58.33333333%;
  }
  .col-md-offset-8{
    margin-left:66.66666667%;
  }
  .col-md-offset-9{
    margin-left:75%;
  }
  .col-md-offset-10{
    margin-left:83.33333333%;
  }
  .col-md-offset-11{
    margin-left:91.66666667%;
  }
  .start-md{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-md{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-md{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-md{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-md{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-md{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-md{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-md{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-md{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-md{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 75em){
  .container{
    width:76rem;
  }
  .col-lg,
  .col-lg-1,
  .col-lg-2,
  .col-lg-3,
  .col-lg-4,
  .col-lg-5,
  .col-lg-6,
  .col-lg-7,
  .col-lg-8,
  .col-lg-9,
  .col-lg-10,
  .col-lg-11,
  .col-lg-12,
  .col-lg-offset-0,
  .col-lg-offset-1,
  .col-lg-offset-2,
  .col-lg-offset-3,
  .col-lg-offset-4,
  .col-lg-offset-5,
  .col-lg-offset-6,
  .col-lg-offset-7,
  .col-lg-offset-8,
  .col-lg-offset-9,
  .col-lg-offset-10,
  .col-lg-offset-11,
  .col-lg-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-lg{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-lg-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-lg-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-lg-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-lg-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-lg-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-lg-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-lg-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-lg-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-lg-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-lg-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-lg-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-lg-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-lg-offset-0{
    margin-left:0;
  }
  .col-lg-offset-1{
    margin-left:8.33333333%;
  }
  .col-lg-offset-2{
    margin-left:16.66666667%;
  }
  .col-lg-offset-3{
    margin-left:25%;
  }
  .col-lg-offset-4{
    margin-left:33.33333333%;
  }
  .col-lg-offset-5{
    margin-left:41.66666667%;
  }
  .col-lg-offset-6{
    margin-left:50%;
  }
  .col-lg-offset-7{
    margin-left:58.33333333%;
  }
  .col-lg-offset-8{
    margin-left:66.66666667%;
  }
  .col-lg-offset-9{
    margin-left:75%;
  }
  .col-lg-offset-10{
    margin-left:83.33333333%;
  }
  .col-lg-offset-11{
    margin-left:91.66666667%;
  }
  .start-lg{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-lg{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-lg{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-lg{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-lg{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-lg{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-lg{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-lg{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-lg{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-lg{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}`]
            },] },
];
/** @nocollapse */
TrainComponent.ctorParameters = () => [
    { type: FormBuilder, },
    { type: StoryService, },
];
TrainComponent.propDecorators = {
    "story": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class StoryComponent {
    /**
     * @param {?} fb
     * @param {?} coreService
     * @param {?} storyService
     */
    constructor(fb, coreService, storyService) {
        this.fb = fb;
        this.coreService = coreService;
        this.storyService = storyService;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.storyTypes = StoryService.storyTypes;
        this.storyTypesArray = Object.keys(this.storyTypes);
        this.storyFormFields = {
            _id: [''],
            storyName: [''],
            intentName: [''],
            speechResponse: [''],
            apiTrigger: [''],
            apiDetails: this.initApiDetails(),
            parameters: this.fb.array(this.story && this.story.parameters ? this.story.parameters.map(n => {
                return this.initParameter(n);
            }) : [])
        };
        this.storyForm = this.fb.group(this.storyFormFields);
        if (this.story) {
            this.coreService.setDataForm(this.storyForm, this.storyFormFields, this.story);
        }
    }
    /**
     * @return {?}
     */
    addParameter() {
        const /** @type {?} */ control = /** @type {?} */ (this.storyForm.controls['parameters']);
        control.push(this.initParameter());
    }
    /**
     * @param {?=} parameter
     * @return {?}
     */
    initParameter(parameter = null) {
        const /** @type {?} */ fields = {
            name: [''],
            type: [''],
            required: [''],
            prompt: ['']
        };
        const /** @type {?} */ g = this.fb.group(fields);
        return g;
    }
    /**
     * @param {?} i
     * @return {?}
     */
    deleteParameter(i) {
        const /** @type {?} */ control = /** @type {?} */ (this.storyForm.controls['parameters']);
        control.removeAt(i);
    }
    /**
     * @param {?=} parameter
     * @return {?}
     */
    initApiDetails(parameter = null) {
        const /** @type {?} */ fields = {
            isJson: [''],
            url: [''],
            requestType: [''],
            jsonData: ['']
        };
        const /** @type {?} */ g = this.fb.group(fields);
        return g;
    }
    /**
     * @return {?}
     */
    save() {
        const /** @type {?} */ form = this.storyForm.value;
        if (form._id && form._id.$oid) {
            form._id = form._id.$oid;
        }
        if (!this.apiTrigger()) {
            delete form.apiDetails;
        }
        this.storyService.saveStory(form)
            .then(c => {
            this.message = 'Story created!';
            if (this.afterSave) {
                this.afterSave(c);
            }
        });
    }
    /**
     * @return {?}
     */
    apiTrigger() {
        return this.storyForm.value.apiTrigger;
    }
}
StoryComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ng-ai-chatbot-story',
                template: `<h2>{{message}}</h2>
<form [formGroup]="storyForm" *ngIf="storyForm" (submit)="save()" class="row">
  <mat-input-container class="col-md-6">
    <textarea matInput formControlName="storyName" placeholder="Story name"></textarea>
  </mat-input-container>
  <mat-input-container class="col-md-6">
    <textarea matInput formControlName="intentName" placeholder="Intent key"></textarea>
  </mat-input-container>
  <h2 class="col-md-12">Parameters
    <button type="button" (click)="addParameter()" mat-button>add</button>
  </h2>
  <section class="col-md-12 parameters" formArrayName="parameters">
    <div *ngFor="let paremeter of storyForm.controls.parameters.controls; let i=index;">
      <div [formGroup]="paremeter" class="row">
        <mat-input-container class="col-md-5">
          <textarea matInput formControlName="name" placeholder="Parameter name"></textarea>
        </mat-input-container>
        <mat-input-container class="col-md-4">
          <mat-select formControlName="type" placeholder="Parameter type">
            <mat-option *ngFor="let story of storyTypesArray" [value]="story">
              {{ storyTypes[story] }}
            </mat-option>
          </mat-select>
        </mat-input-container>
        <mat-checkbox class="col-md-2" formControlName="required">Required</mat-checkbox>
        <mat-input-container class="col-md-10" *ngIf="paremeter.controls.required.value">
          <textarea matInput formControlName="prompt" placeholder="Message to prompt when require"></textarea>
        </mat-input-container>
        <button type="button" class="col-md-1" (click)="deleteParameter(i)" mat-button>delete</button>
      </div>
    </div>
  </section>
  <h2 class="col-md-12">
    <mat-checkbox formControlName="apiTrigger">API trigger</mat-checkbox>
  </h2>
  <section class="row" [formGroup]="storyForm.controls.apiDetails" *ngIf="apiTrigger()">
    <mat-checkbox class="col-md-2" formControlName="isJson">JSON</mat-checkbox>
    <mat-input-container class="col-md-6">
      <textarea matInput formControlName="url" placeholder="API url"></textarea>
    </mat-input-container>
    <mat-input-container class="col-md-2">
      <mat-select formControlName="requestType" placeholder="API method">
        <mat-option value="GET">GET</mat-option>
        <mat-option value="POST">POST</mat-option>
        <mat-option value="PUT">PUT</mat-option>
        <mat-option value="DELETE">DELETE</mat-option>
      </mat-select>
    </mat-input-container>
    <mat-input-container class="col-md-12">
      <textarea matInput formControlName="jsonData" placeholder="JsonData"></textarea>
    </mat-input-container>
  </section>
  <mat-input-container class="col-md-12">
    <textarea matInput formControlName="speechResponse" placeholder="Intent key"></textarea>
  </mat-input-container>
  <p>
    Extracted parameters and api call response can be accessed from speech respone using "parameters" and "result" objects respectively
  </p>
  <button class="col-md-1" mat-button>Save</button>
</form>`,
                styles: [`
section.parameters{
    overflow-y:auto;
    height:200px;
}`, `.container-fluid,
.container{
  margin-right:auto;
  margin-left:auto;
}
.container-fluid{
  padding-right:2rem;
  padding-left:2rem;
}
.row{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  display:-webkit-box;
  display:-ms-flexbox;
  display:flex;
  -webkit-box-flex:0;
  -ms-flex:0 1 auto;
  flex:0 1 auto;
  -webkit-box-orient:horizontal;
  -webkit-box-direction:normal;
  -ms-flex-direction:row;
  flex-direction:row;
  -ms-flex-wrap:wrap;
  flex-wrap:wrap;
  margin-right:-0.5rem;
  margin-left:-0.5rem;
}
.row.reverse{
  -webkit-box-orient:horizontal;
  -webkit-box-direction:reverse;
  -ms-flex-direction:row-reverse;
  flex-direction:row-reverse;
}
.col.reverse{
  -webkit-box-orient:vertical;
  -webkit-box-direction:reverse;
  -ms-flex-direction:column-reverse;
  flex-direction:column-reverse;
}
.col-xs,
.col-xs-1,
.col-xs-2,
.col-xs-3,
.col-xs-4,
.col-xs-5,
.col-xs-6,
.col-xs-7,
.col-xs-8,
.col-xs-9,
.col-xs-10,
.col-xs-11,
.col-xs-12,
.col-xs-offset-0,
.col-xs-offset-1,
.col-xs-offset-2,
.col-xs-offset-3,
.col-xs-offset-4,
.col-xs-offset-5,
.col-xs-offset-6,
.col-xs-offset-7,
.col-xs-offset-8,
.col-xs-offset-9,
.col-xs-offset-10,
.col-xs-offset-11,
.col-xs-offset-12{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  -webkit-box-flex:0;
  -ms-flex:0 0 auto;
  flex:0 0 auto;
  padding-right:0.5rem;
  padding-left:0.5rem;
}
.col-xs{
  -webkit-box-flex:1;
  -ms-flex-positive:1;
  flex-grow:1;
  -ms-flex-preferred-size:0;
  flex-basis:0;
  max-width:100%;
}
.col-xs-1{
  -ms-flex-preferred-size:8.33333333%;
  flex-basis:8.33333333%;
  max-width:8.33333333%;
}
.col-xs-2{
  -ms-flex-preferred-size:16.66666667%;
  flex-basis:16.66666667%;
  max-width:16.66666667%;
}
.col-xs-3{
  -ms-flex-preferred-size:25%;
  flex-basis:25%;
  max-width:25%;
}
.col-xs-4{
  -ms-flex-preferred-size:33.33333333%;
  flex-basis:33.33333333%;
  max-width:33.33333333%;
}
.col-xs-5{
  -ms-flex-preferred-size:41.66666667%;
  flex-basis:41.66666667%;
  max-width:41.66666667%;
}
.col-xs-6{
  -ms-flex-preferred-size:50%;
  flex-basis:50%;
  max-width:50%;
}
.col-xs-7{
  -ms-flex-preferred-size:58.33333333%;
  flex-basis:58.33333333%;
  max-width:58.33333333%;
}
.col-xs-8{
  -ms-flex-preferred-size:66.66666667%;
  flex-basis:66.66666667%;
  max-width:66.66666667%;
}
.col-xs-9{
  -ms-flex-preferred-size:75%;
  flex-basis:75%;
  max-width:75%;
}
.col-xs-10{
  -ms-flex-preferred-size:83.33333333%;
  flex-basis:83.33333333%;
  max-width:83.33333333%;
}
.col-xs-11{
  -ms-flex-preferred-size:91.66666667%;
  flex-basis:91.66666667%;
  max-width:91.66666667%;
}
.col-xs-12{
  -ms-flex-preferred-size:100%;
  flex-basis:100%;
  max-width:100%;
}
.col-xs-offset-0{
  margin-left:0;
}
.col-xs-offset-1{
  margin-left:8.33333333%;
}
.col-xs-offset-2{
  margin-left:16.66666667%;
}
.col-xs-offset-3{
  margin-left:25%;
}
.col-xs-offset-4{
  margin-left:33.33333333%;
}
.col-xs-offset-5{
  margin-left:41.66666667%;
}
.col-xs-offset-6{
  margin-left:50%;
}
.col-xs-offset-7{
  margin-left:58.33333333%;
}
.col-xs-offset-8{
  margin-left:66.66666667%;
}
.col-xs-offset-9{
  margin-left:75%;
}
.col-xs-offset-10{
  margin-left:83.33333333%;
}
.col-xs-offset-11{
  margin-left:91.66666667%;
}
.start-xs{
  -webkit-box-pack:start;
  -ms-flex-pack:start;
  justify-content:flex-start;
  text-align:start;
}
.center-xs{
  -webkit-box-pack:center;
  -ms-flex-pack:center;
  justify-content:center;
  text-align:center;
}
.end-xs{
  -webkit-box-pack:end;
  -ms-flex-pack:end;
  justify-content:flex-end;
  text-align:end;
}
.top-xs{
  -webkit-box-align:start;
  -ms-flex-align:start;
  align-items:flex-start;
}
.middle-xs{
  -webkit-box-align:center;
  -ms-flex-align:center;
  align-items:center;
}
.bottom-xs{
  -webkit-box-align:end;
  -ms-flex-align:end;
  align-items:flex-end;
}
.around-xs{
  -ms-flex-pack:distribute;
  justify-content:space-around;
}
.between-xs{
  -webkit-box-pack:justify;
  -ms-flex-pack:justify;
  justify-content:space-between;
}
.first-xs{
  -webkit-box-ordinal-group:0;
  -ms-flex-order:-1;
  order:-1;
}
.last-xs{
  -webkit-box-ordinal-group:2;
  -ms-flex-order:1;
  order:1;
}
@media only screen and (min-width: 48em){
  .container{
    width:49rem;
  }
  .col-sm,
  .col-sm-1,
  .col-sm-2,
  .col-sm-3,
  .col-sm-4,
  .col-sm-5,
  .col-sm-6,
  .col-sm-7,
  .col-sm-8,
  .col-sm-9,
  .col-sm-10,
  .col-sm-11,
  .col-sm-12,
  .col-sm-offset-0,
  .col-sm-offset-1,
  .col-sm-offset-2,
  .col-sm-offset-3,
  .col-sm-offset-4,
  .col-sm-offset-5,
  .col-sm-offset-6,
  .col-sm-offset-7,
  .col-sm-offset-8,
  .col-sm-offset-9,
  .col-sm-offset-10,
  .col-sm-offset-11,
  .col-sm-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-sm{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-sm-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-sm-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-sm-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-sm-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-sm-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-sm-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-sm-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-sm-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-sm-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-sm-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-sm-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-sm-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-sm-offset-0{
    margin-left:0;
  }
  .col-sm-offset-1{
    margin-left:8.33333333%;
  }
  .col-sm-offset-2{
    margin-left:16.66666667%;
  }
  .col-sm-offset-3{
    margin-left:25%;
  }
  .col-sm-offset-4{
    margin-left:33.33333333%;
  }
  .col-sm-offset-5{
    margin-left:41.66666667%;
  }
  .col-sm-offset-6{
    margin-left:50%;
  }
  .col-sm-offset-7{
    margin-left:58.33333333%;
  }
  .col-sm-offset-8{
    margin-left:66.66666667%;
  }
  .col-sm-offset-9{
    margin-left:75%;
  }
  .col-sm-offset-10{
    margin-left:83.33333333%;
  }
  .col-sm-offset-11{
    margin-left:91.66666667%;
  }
  .start-sm{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-sm{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-sm{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-sm{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-sm{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-sm{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-sm{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-sm{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-sm{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-sm{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 64em){
  .container{
    width:65rem;
  }
  .col-md,
  .col-md-1,
  .col-md-2,
  .col-md-3,
  .col-md-4,
  .col-md-5,
  .col-md-6,
  .col-md-7,
  .col-md-8,
  .col-md-9,
  .col-md-10,
  .col-md-11,
  .col-md-12,
  .col-md-offset-0,
  .col-md-offset-1,
  .col-md-offset-2,
  .col-md-offset-3,
  .col-md-offset-4,
  .col-md-offset-5,
  .col-md-offset-6,
  .col-md-offset-7,
  .col-md-offset-8,
  .col-md-offset-9,
  .col-md-offset-10,
  .col-md-offset-11,
  .col-md-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-md{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-md-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-md-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-md-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-md-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-md-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-md-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-md-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-md-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-md-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-md-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-md-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-md-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-md-offset-0{
    margin-left:0;
  }
  .col-md-offset-1{
    margin-left:8.33333333%;
  }
  .col-md-offset-2{
    margin-left:16.66666667%;
  }
  .col-md-offset-3{
    margin-left:25%;
  }
  .col-md-offset-4{
    margin-left:33.33333333%;
  }
  .col-md-offset-5{
    margin-left:41.66666667%;
  }
  .col-md-offset-6{
    margin-left:50%;
  }
  .col-md-offset-7{
    margin-left:58.33333333%;
  }
  .col-md-offset-8{
    margin-left:66.66666667%;
  }
  .col-md-offset-9{
    margin-left:75%;
  }
  .col-md-offset-10{
    margin-left:83.33333333%;
  }
  .col-md-offset-11{
    margin-left:91.66666667%;
  }
  .start-md{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-md{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-md{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-md{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-md{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-md{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-md{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-md{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-md{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-md{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 75em){
  .container{
    width:76rem;
  }
  .col-lg,
  .col-lg-1,
  .col-lg-2,
  .col-lg-3,
  .col-lg-4,
  .col-lg-5,
  .col-lg-6,
  .col-lg-7,
  .col-lg-8,
  .col-lg-9,
  .col-lg-10,
  .col-lg-11,
  .col-lg-12,
  .col-lg-offset-0,
  .col-lg-offset-1,
  .col-lg-offset-2,
  .col-lg-offset-3,
  .col-lg-offset-4,
  .col-lg-offset-5,
  .col-lg-offset-6,
  .col-lg-offset-7,
  .col-lg-offset-8,
  .col-lg-offset-9,
  .col-lg-offset-10,
  .col-lg-offset-11,
  .col-lg-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-lg{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-lg-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-lg-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-lg-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-lg-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-lg-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-lg-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-lg-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-lg-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-lg-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-lg-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-lg-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-lg-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-lg-offset-0{
    margin-left:0;
  }
  .col-lg-offset-1{
    margin-left:8.33333333%;
  }
  .col-lg-offset-2{
    margin-left:16.66666667%;
  }
  .col-lg-offset-3{
    margin-left:25%;
  }
  .col-lg-offset-4{
    margin-left:33.33333333%;
  }
  .col-lg-offset-5{
    margin-left:41.66666667%;
  }
  .col-lg-offset-6{
    margin-left:50%;
  }
  .col-lg-offset-7{
    margin-left:58.33333333%;
  }
  .col-lg-offset-8{
    margin-left:66.66666667%;
  }
  .col-lg-offset-9{
    margin-left:75%;
  }
  .col-lg-offset-10{
    margin-left:83.33333333%;
  }
  .col-lg-offset-11{
    margin-left:91.66666667%;
  }
  .start-lg{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-lg{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-lg{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-lg{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-lg{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-lg{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-lg{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-lg{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-lg{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-lg{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}`]
            },] },
];
/** @nocollapse */
StoryComponent.ctorParameters = () => [
    { type: FormBuilder, },
    { type: CoreService, },
    { type: StoryService, },
];
StoryComponent.propDecorators = {
    "afterSave": [{ type: Input },],
    "story": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class BotsComponent {
    /**
     * @param {?} dialog
     * @param {?} fb
     * @param {?} storyService
     */
    constructor(dialog, fb, storyService) {
        this.dialog = dialog;
        this.fb = fb;
        this.storyService = storyService;
        this.showAddForm = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.botFormFields = {
            _id: [''],
            botName: [''],
            username: [''],
            password: ['']
        };
        this.botForm = this.fb.group(this.botFormFields);
        this.storyService.getBots().then((s) => {
            this.bots = s;
            this.bots.push({ botName: 'default' });
        });
    }
    /**
     * @param {?} bot
     * @return {?}
     */
    stories(bot) {
        if (this.storiesBot) {
            this.storiesBot(bot);
        }
    }
    /**
     * @param {?} bot
     * @return {?}
     */
    edit(bot) {
    }
    /**
     * @return {?}
     */
    login() {
        const /** @type {?} */ form = this.botForm.value;
        localStorage["botToken"] = btoa(`${form.username}:${form.password}`);
        this.storyService.getBots()
            .then(c => {
            this.bots = c;
        });
    }
    /**
     * @return {?}
     */
    save() {
        const /** @type {?} */ form = this.botForm.value;
        this.storyService.saveBot(form)
            .then(c => {
            if (this.afterSave) {
                this.afterSave(c);
            }
            this.ngOnInit();
            this.showAddForm = false;
        })
            .catch(c => {
            this.message = 'Error saving bot';
        });
    }
}
BotsComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line:component-selector
                selector: 'ng-ai-chatbot-bots',
                template: `<h1>
  Bots
  <button mat-mini-fab (click)="showAddForm=true">
    <mat-icon aria-label="Example icon-button with a heart icon">add</mat-icon>
  </button>
</h1>
<form [formGroup]="botForm" (submit)="save()" class="row" *ngIf="botForm && showAddForm">
  {{message}}
  <mat-input-container class="col-md-4">
    <textarea matInput formControlName="botName" placeholder="Bot name"></textarea>
  </mat-input-container>
  <mat-input-container class="col-md-4">
    <input matInput formControlName="username" placeholder="username"/>
  </mat-input-container>
  <mat-input-container class="col-md-3">
    <input type="password" matInput formControlName="password" placeholder="password"/>
  </mat-input-container>
  <button class="col-md-1" mat-button>Save</button>
</form>
<form [formGroup]="botForm" (submit)="login()" class="row" *ngIf="botForm && !showAddForm">
    <h3>Login your bot</h3>
    <mat-input-container class="col-md-4">
      <input matInput formControlName="username" placeholder="username"/>
    </mat-input-container>
    <mat-input-container class="col-md-3">
      <input  type="password" matInput formControlName="password" placeholder="password"/>
    </mat-input-container>
    <button class="col-md-1" mat-button>Save</button>
  </form>
<div class="row">
  <mat-card class="col-md-6" *ngFor="let bot of bots">
    <mat-card-header>
      <div mat-card-avatar class="example-header-image"></div>
      <mat-card-title>{{bot.botName}}</mat-card-title>
    </mat-card-header>
    <mat-card-actions>
      <button mat-button (click)="edit(bot)" *ngIf="bot._id">Edit</button>
      <button mat-button (click)="stories(bot)">Stories</button>
    </mat-card-actions>
  </mat-card>
</div>`,
                styles: [``, `.container-fluid,
.container{
  margin-right:auto;
  margin-left:auto;
}
.container-fluid{
  padding-right:2rem;
  padding-left:2rem;
}
.row{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  display:-webkit-box;
  display:-ms-flexbox;
  display:flex;
  -webkit-box-flex:0;
  -ms-flex:0 1 auto;
  flex:0 1 auto;
  -webkit-box-orient:horizontal;
  -webkit-box-direction:normal;
  -ms-flex-direction:row;
  flex-direction:row;
  -ms-flex-wrap:wrap;
  flex-wrap:wrap;
  margin-right:-0.5rem;
  margin-left:-0.5rem;
}
.row.reverse{
  -webkit-box-orient:horizontal;
  -webkit-box-direction:reverse;
  -ms-flex-direction:row-reverse;
  flex-direction:row-reverse;
}
.col.reverse{
  -webkit-box-orient:vertical;
  -webkit-box-direction:reverse;
  -ms-flex-direction:column-reverse;
  flex-direction:column-reverse;
}
.col-xs,
.col-xs-1,
.col-xs-2,
.col-xs-3,
.col-xs-4,
.col-xs-5,
.col-xs-6,
.col-xs-7,
.col-xs-8,
.col-xs-9,
.col-xs-10,
.col-xs-11,
.col-xs-12,
.col-xs-offset-0,
.col-xs-offset-1,
.col-xs-offset-2,
.col-xs-offset-3,
.col-xs-offset-4,
.col-xs-offset-5,
.col-xs-offset-6,
.col-xs-offset-7,
.col-xs-offset-8,
.col-xs-offset-9,
.col-xs-offset-10,
.col-xs-offset-11,
.col-xs-offset-12{
  -webkit-box-sizing:border-box;
          box-sizing:border-box;
  -webkit-box-flex:0;
  -ms-flex:0 0 auto;
  flex:0 0 auto;
  padding-right:0.5rem;
  padding-left:0.5rem;
}
.col-xs{
  -webkit-box-flex:1;
  -ms-flex-positive:1;
  flex-grow:1;
  -ms-flex-preferred-size:0;
  flex-basis:0;
  max-width:100%;
}
.col-xs-1{
  -ms-flex-preferred-size:8.33333333%;
  flex-basis:8.33333333%;
  max-width:8.33333333%;
}
.col-xs-2{
  -ms-flex-preferred-size:16.66666667%;
  flex-basis:16.66666667%;
  max-width:16.66666667%;
}
.col-xs-3{
  -ms-flex-preferred-size:25%;
  flex-basis:25%;
  max-width:25%;
}
.col-xs-4{
  -ms-flex-preferred-size:33.33333333%;
  flex-basis:33.33333333%;
  max-width:33.33333333%;
}
.col-xs-5{
  -ms-flex-preferred-size:41.66666667%;
  flex-basis:41.66666667%;
  max-width:41.66666667%;
}
.col-xs-6{
  -ms-flex-preferred-size:50%;
  flex-basis:50%;
  max-width:50%;
}
.col-xs-7{
  -ms-flex-preferred-size:58.33333333%;
  flex-basis:58.33333333%;
  max-width:58.33333333%;
}
.col-xs-8{
  -ms-flex-preferred-size:66.66666667%;
  flex-basis:66.66666667%;
  max-width:66.66666667%;
}
.col-xs-9{
  -ms-flex-preferred-size:75%;
  flex-basis:75%;
  max-width:75%;
}
.col-xs-10{
  -ms-flex-preferred-size:83.33333333%;
  flex-basis:83.33333333%;
  max-width:83.33333333%;
}
.col-xs-11{
  -ms-flex-preferred-size:91.66666667%;
  flex-basis:91.66666667%;
  max-width:91.66666667%;
}
.col-xs-12{
  -ms-flex-preferred-size:100%;
  flex-basis:100%;
  max-width:100%;
}
.col-xs-offset-0{
  margin-left:0;
}
.col-xs-offset-1{
  margin-left:8.33333333%;
}
.col-xs-offset-2{
  margin-left:16.66666667%;
}
.col-xs-offset-3{
  margin-left:25%;
}
.col-xs-offset-4{
  margin-left:33.33333333%;
}
.col-xs-offset-5{
  margin-left:41.66666667%;
}
.col-xs-offset-6{
  margin-left:50%;
}
.col-xs-offset-7{
  margin-left:58.33333333%;
}
.col-xs-offset-8{
  margin-left:66.66666667%;
}
.col-xs-offset-9{
  margin-left:75%;
}
.col-xs-offset-10{
  margin-left:83.33333333%;
}
.col-xs-offset-11{
  margin-left:91.66666667%;
}
.start-xs{
  -webkit-box-pack:start;
  -ms-flex-pack:start;
  justify-content:flex-start;
  text-align:start;
}
.center-xs{
  -webkit-box-pack:center;
  -ms-flex-pack:center;
  justify-content:center;
  text-align:center;
}
.end-xs{
  -webkit-box-pack:end;
  -ms-flex-pack:end;
  justify-content:flex-end;
  text-align:end;
}
.top-xs{
  -webkit-box-align:start;
  -ms-flex-align:start;
  align-items:flex-start;
}
.middle-xs{
  -webkit-box-align:center;
  -ms-flex-align:center;
  align-items:center;
}
.bottom-xs{
  -webkit-box-align:end;
  -ms-flex-align:end;
  align-items:flex-end;
}
.around-xs{
  -ms-flex-pack:distribute;
  justify-content:space-around;
}
.between-xs{
  -webkit-box-pack:justify;
  -ms-flex-pack:justify;
  justify-content:space-between;
}
.first-xs{
  -webkit-box-ordinal-group:0;
  -ms-flex-order:-1;
  order:-1;
}
.last-xs{
  -webkit-box-ordinal-group:2;
  -ms-flex-order:1;
  order:1;
}
@media only screen and (min-width: 48em){
  .container{
    width:49rem;
  }
  .col-sm,
  .col-sm-1,
  .col-sm-2,
  .col-sm-3,
  .col-sm-4,
  .col-sm-5,
  .col-sm-6,
  .col-sm-7,
  .col-sm-8,
  .col-sm-9,
  .col-sm-10,
  .col-sm-11,
  .col-sm-12,
  .col-sm-offset-0,
  .col-sm-offset-1,
  .col-sm-offset-2,
  .col-sm-offset-3,
  .col-sm-offset-4,
  .col-sm-offset-5,
  .col-sm-offset-6,
  .col-sm-offset-7,
  .col-sm-offset-8,
  .col-sm-offset-9,
  .col-sm-offset-10,
  .col-sm-offset-11,
  .col-sm-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-sm{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-sm-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-sm-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-sm-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-sm-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-sm-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-sm-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-sm-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-sm-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-sm-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-sm-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-sm-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-sm-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-sm-offset-0{
    margin-left:0;
  }
  .col-sm-offset-1{
    margin-left:8.33333333%;
  }
  .col-sm-offset-2{
    margin-left:16.66666667%;
  }
  .col-sm-offset-3{
    margin-left:25%;
  }
  .col-sm-offset-4{
    margin-left:33.33333333%;
  }
  .col-sm-offset-5{
    margin-left:41.66666667%;
  }
  .col-sm-offset-6{
    margin-left:50%;
  }
  .col-sm-offset-7{
    margin-left:58.33333333%;
  }
  .col-sm-offset-8{
    margin-left:66.66666667%;
  }
  .col-sm-offset-9{
    margin-left:75%;
  }
  .col-sm-offset-10{
    margin-left:83.33333333%;
  }
  .col-sm-offset-11{
    margin-left:91.66666667%;
  }
  .start-sm{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-sm{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-sm{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-sm{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-sm{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-sm{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-sm{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-sm{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-sm{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-sm{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 64em){
  .container{
    width:65rem;
  }
  .col-md,
  .col-md-1,
  .col-md-2,
  .col-md-3,
  .col-md-4,
  .col-md-5,
  .col-md-6,
  .col-md-7,
  .col-md-8,
  .col-md-9,
  .col-md-10,
  .col-md-11,
  .col-md-12,
  .col-md-offset-0,
  .col-md-offset-1,
  .col-md-offset-2,
  .col-md-offset-3,
  .col-md-offset-4,
  .col-md-offset-5,
  .col-md-offset-6,
  .col-md-offset-7,
  .col-md-offset-8,
  .col-md-offset-9,
  .col-md-offset-10,
  .col-md-offset-11,
  .col-md-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-md{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-md-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-md-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-md-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-md-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-md-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-md-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-md-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-md-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-md-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-md-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-md-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-md-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-md-offset-0{
    margin-left:0;
  }
  .col-md-offset-1{
    margin-left:8.33333333%;
  }
  .col-md-offset-2{
    margin-left:16.66666667%;
  }
  .col-md-offset-3{
    margin-left:25%;
  }
  .col-md-offset-4{
    margin-left:33.33333333%;
  }
  .col-md-offset-5{
    margin-left:41.66666667%;
  }
  .col-md-offset-6{
    margin-left:50%;
  }
  .col-md-offset-7{
    margin-left:58.33333333%;
  }
  .col-md-offset-8{
    margin-left:66.66666667%;
  }
  .col-md-offset-9{
    margin-left:75%;
  }
  .col-md-offset-10{
    margin-left:83.33333333%;
  }
  .col-md-offset-11{
    margin-left:91.66666667%;
  }
  .start-md{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-md{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-md{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-md{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-md{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-md{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-md{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-md{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-md{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-md{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}
@media only screen and (min-width: 75em){
  .container{
    width:76rem;
  }
  .col-lg,
  .col-lg-1,
  .col-lg-2,
  .col-lg-3,
  .col-lg-4,
  .col-lg-5,
  .col-lg-6,
  .col-lg-7,
  .col-lg-8,
  .col-lg-9,
  .col-lg-10,
  .col-lg-11,
  .col-lg-12,
  .col-lg-offset-0,
  .col-lg-offset-1,
  .col-lg-offset-2,
  .col-lg-offset-3,
  .col-lg-offset-4,
  .col-lg-offset-5,
  .col-lg-offset-6,
  .col-lg-offset-7,
  .col-lg-offset-8,
  .col-lg-offset-9,
  .col-lg-offset-10,
  .col-lg-offset-11,
  .col-lg-offset-12{
    -webkit-box-sizing:border-box;
            box-sizing:border-box;
    -webkit-box-flex:0;
    -ms-flex:0 0 auto;
    flex:0 0 auto;
    padding-right:0.5rem;
    padding-left:0.5rem;
  }
  .col-lg{
    -webkit-box-flex:1;
    -ms-flex-positive:1;
    flex-grow:1;
    -ms-flex-preferred-size:0;
    flex-basis:0;
    max-width:100%;
  }
  .col-lg-1{
    -ms-flex-preferred-size:8.33333333%;
    flex-basis:8.33333333%;
    max-width:8.33333333%;
  }
  .col-lg-2{
    -ms-flex-preferred-size:16.66666667%;
    flex-basis:16.66666667%;
    max-width:16.66666667%;
  }
  .col-lg-3{
    -ms-flex-preferred-size:25%;
    flex-basis:25%;
    max-width:25%;
  }
  .col-lg-4{
    -ms-flex-preferred-size:33.33333333%;
    flex-basis:33.33333333%;
    max-width:33.33333333%;
  }
  .col-lg-5{
    -ms-flex-preferred-size:41.66666667%;
    flex-basis:41.66666667%;
    max-width:41.66666667%;
  }
  .col-lg-6{
    -ms-flex-preferred-size:50%;
    flex-basis:50%;
    max-width:50%;
  }
  .col-lg-7{
    -ms-flex-preferred-size:58.33333333%;
    flex-basis:58.33333333%;
    max-width:58.33333333%;
  }
  .col-lg-8{
    -ms-flex-preferred-size:66.66666667%;
    flex-basis:66.66666667%;
    max-width:66.66666667%;
  }
  .col-lg-9{
    -ms-flex-preferred-size:75%;
    flex-basis:75%;
    max-width:75%;
  }
  .col-lg-10{
    -ms-flex-preferred-size:83.33333333%;
    flex-basis:83.33333333%;
    max-width:83.33333333%;
  }
  .col-lg-11{
    -ms-flex-preferred-size:91.66666667%;
    flex-basis:91.66666667%;
    max-width:91.66666667%;
  }
  .col-lg-12{
    -ms-flex-preferred-size:100%;
    flex-basis:100%;
    max-width:100%;
  }
  .col-lg-offset-0{
    margin-left:0;
  }
  .col-lg-offset-1{
    margin-left:8.33333333%;
  }
  .col-lg-offset-2{
    margin-left:16.66666667%;
  }
  .col-lg-offset-3{
    margin-left:25%;
  }
  .col-lg-offset-4{
    margin-left:33.33333333%;
  }
  .col-lg-offset-5{
    margin-left:41.66666667%;
  }
  .col-lg-offset-6{
    margin-left:50%;
  }
  .col-lg-offset-7{
    margin-left:58.33333333%;
  }
  .col-lg-offset-8{
    margin-left:66.66666667%;
  }
  .col-lg-offset-9{
    margin-left:75%;
  }
  .col-lg-offset-10{
    margin-left:83.33333333%;
  }
  .col-lg-offset-11{
    margin-left:91.66666667%;
  }
  .start-lg{
    -webkit-box-pack:start;
    -ms-flex-pack:start;
    justify-content:flex-start;
    text-align:start;
  }
  .center-lg{
    -webkit-box-pack:center;
    -ms-flex-pack:center;
    justify-content:center;
    text-align:center;
  }
  .end-lg{
    -webkit-box-pack:end;
    -ms-flex-pack:end;
    justify-content:flex-end;
    text-align:end;
  }
  .top-lg{
    -webkit-box-align:start;
    -ms-flex-align:start;
    align-items:flex-start;
  }
  .middle-lg{
    -webkit-box-align:center;
    -ms-flex-align:center;
    align-items:center;
  }
  .bottom-lg{
    -webkit-box-align:end;
    -ms-flex-align:end;
    align-items:flex-end;
  }
  .around-lg{
    -ms-flex-pack:distribute;
    justify-content:space-around;
  }
  .between-lg{
    -webkit-box-pack:justify;
    -ms-flex-pack:justify;
    justify-content:space-between;
  }
  .first-lg{
    -webkit-box-ordinal-group:0;
    -ms-flex-order:-1;
    order:-1;
  }
  .last-lg{
    -webkit-box-ordinal-group:2;
    -ms-flex-order:1;
    order:1;
  }
}`]
            },] },
];
/** @nocollapse */
BotsComponent.ctorParameters = () => [
    { type: MatDialog, },
    { type: FormBuilder, },
    { type: StoryService, },
];
BotsComponent.propDecorators = {
    "storiesBot": [{ type: Input },],
    "afterSave": [{ type: Input },],
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class NgAiChatbot {
    /**
     * @return {?}
     */
    static forRoot() {
        return {
            ngModule: NgAiChatbot,
            providers: []
        };
    }
}
NgAiChatbot.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HttpClientModule,
                    BrowserAnimationsModule,
                    ReactiveFormsModule,
                    MatInputModule, MatCardModule, MatButtonModule, MatDialogModule,
                    MatCheckboxModule, MatIconModule, MatListModule, MatGridListModule, MatSelectModule,
                    MatToolbarModule, MatIconModule, MatFormFieldModule
                ],
                declarations: [
                    ChatComponent, StoriesComponent, StoryComponent, TrainComponent,
                    StoryDialogComponent, TrainDialogComponent, BotsComponent
                ],
                exports: [
                    ChatComponent, StoriesComponent, StoryComponent, TrainComponent,
                    StoryDialogComponent, TrainDialogComponent, BotsComponent
                ],
                providers: [CoreService, StoryService, ChatService],
                entryComponents: [StoryDialogComponent, TrainDialogComponent, BotsComponent]
            },] },
];
/** @nocollapse */
NgAiChatbot.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { NgAiChatbot, ChatComponent, StoriesComponent, TrainComponent, StoryComponent, ChatService as a, CoreService as b, BotsComponent as f, StoryService as c, StoryDialogComponent as d, TrainDialogComponent as e };
